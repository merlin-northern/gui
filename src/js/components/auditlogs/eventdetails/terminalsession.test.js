import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { defaultState, undefineds } from '../../../../../tests/mockData';
import TerminalSession from './terminalsession';

const mockStore = configureStore([thunk]);

describe('TerminalSession Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Provider store={store}>
          <TerminalSession item={defaultState.organization.events[2]} />
        </Provider>
      </MemoryRouter>
    );

    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
