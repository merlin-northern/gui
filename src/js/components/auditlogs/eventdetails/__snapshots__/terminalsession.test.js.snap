// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerminalSession Component renders correctly 1`] = `
<div
  class="loaderContainer"
>
  <div
    class="loader"
  >
    <span
      class="dot dot_1"
    />
    <span
      class="dot dot_2"
    />
    <span
      class="dot dot_3"
    />
    <span
      class="dot dot_4"
    />
  </div>
</div>
`;
